tests/animalShelter.test.ts

```diff
@@ -106,29 +106,45 @@ describe('Animal Shelter Tests', () => {
-    const animalShelter = AnimalShelter.getInstance();
-    animalShelter.setShelterTerritory(2000);
-    expect(animalShelter.getShelterTerritory()).toBe(2000);
-  });
-  test("should throw an error if shelter trritory set to negative", () => {
-    const animalShelter = AnimalShelter.getInstance();
-    expect(() => { animalShelter.setShelterTerritory(-2000) }).toThrow(NEGATIVE_SHELTER_TERRITORY);
-  });
+  test('should succesfully set shelter trritory', () => {
+    const animalShelter = AnimalShelter.getInstance()
+    animalShelter.setShelterTerritory(2000)
+    expect(animalShelter.getShelterTerritory()).toBe(2000)
+  })
+  test('should throw an error if shelter trritory set to negative', () => {
+    const animalShelter = AnimalShelter.getInstance()
+    expect(() => {
+      animalShelter.setShelterTerritory(-2000)
+    }).toThrow(NEGATIVE_SHELTER_TERRITORY)
+  })
 
-  test("report should return an object with the correct number of animals with specific abilities", () => {
-    const animalShelter = AnimalShelter.getInstance();
-    const tortoise = new Tortoise("t1", 3, false);
-    const tortoise2 = new Tortoise("t2", 4, false);
-    const squirrel = new Squirrel("s1", 3, false, "Maple", 44, 4);
-    const lion = new Lion("simba", 4, true);
-    animalShelter.addAnimal(tortoise);
-    animalShelter.addAnimal(tortoise2);
-    animalShelter.addAnimal(squirrel);
-    animalShelter.addAnimal(lion);
-    const animals = animalShelter.report();
-    expect(animals.anaimalsWhoClimbTrees).toBe(4);
-    expect(animals.animalsWhoJump).toBe(2);
-    expect(animals.animalsWhoRun).toBe(1);
-  });
+  test('report should return an object with the correct number of animals with specific abilities', () => {
+    const animalShelter = AnimalShelter.getInstance()
+    const tortoise = new Tortoise('t1', 3, false)
+    const tortoise2 = new Tortoise('t2', 4, false)
+    const squirrel = new Squirrel('s1', 3, false, 'Maple', 44, 4)
+    const lion = new Lion('simba', 4, true)
+    animalShelter.addAnimal(tortoise)
+    animalShelter.addAnimal(tortoise2)
+    animalShelter.addAnimal(squirrel)
+    animalShelter.addAnimal(lion)
+    const animals = animalShelter.report()
+    expect(animals.anaimalsWhoClimbTrees).toBe(4)
+    expect(animals.animalsWhoJump).toBe(2)
+    expect(animals.animalsWhoRun).toBe(1)
+  })
+})
+test('should return an estimate of future space needed', () => {
+  const animalShelter = AnimalShelter.getInstance()
+  const tortoise = new Tortoise('t1', 3, false)
+  const tortoise2 = new Tortoise('t2', 4, false)
+  const squirrel = new Squirrel('s1', 3, false, 'Maple', 44, 4)
+  const squirrel2 = new Squirrel('s2', 4, false, 'Maple', 44, 4)
+  const lion = new Lion('simba', 4, true)
+  animalShelter.addAnimal(tortoise)
+  animalShelter.addAnimal(tortoise2)
+  animalShelter.addAnimal(squirrel)
+  animalShelter.addAnimal(squirrel2)
+  animalShelter.addAnimal(lion)
+  const estimate = animalShelter.getFutureEstimate(2)
```

Comment:
what are  we checking here ? i dont see anything being tested.    Make additional tests and try to cover all corner cases

Reply:
Yeah not really sure how to do so because we are using random and the estimate that is returned changes every time

Reply:
If you have such issue , dont make a pr but ask colleagues for help or even better do some research on the topic. I see you found a way to do this already, but in the future don't let out pr's that are not finished without asking explicitly for help on the troublesome topic